test/bio_socket_test.c

```diff
@@ -0,0 +1,167 @@
+/*
+ * Copyright 2022 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+#include <stdio.h>
+#include <string.h>
+#include <sys/types.h>
+#include <netinet/in.h>
+#include <arpa/inet.h>
+#include <fcntl.h>
+#include <openssl/bio.h>
+#include <netinet/tcp.h>
```

Comment:
Are all these available on all platforms?

Reply:
I pushed this so I can use the checks to validate other platforms.

Reply:
And... certainly not on windows...

Reply:
Fixed